### PR TITLE
[Bug]: Set due date setting is  cannot be searched

### LINKED ISSUE #16603: [BUG]: "Set due date" setting in App bar buttons cannot be searched along with "Reset progress"
### Checked for duplicates?

- [X] This issue is not a duplicate

### Does it also happen in the desktop version?

- [ ] This bug does not occur in the latest version of Anki

### What are the steps to reproduce this bug?

hamburger menu —> settings —> search icon —> put text `Set due date` and `Reset progress`

### Expected behaviour

search works

### Actual behaviour

search doesn't work

### Debug info

```text
latest alpha
```

### (Optional) Anything else you want to share?

_No response_

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
## Purpose / Description
Set due date setting is  cannot be searched

## Fixes
* Fixes #16603 

## How Has This Been Tested?

physical device SDK 34

##Screenshot
WhatsApp Image 2024-10-07 at 10 02 12 PM

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/res/values/10-preferences.xml [hunk 1]
```diff
@@ -181,6 +181,8 @@
     <string name="pref_cat_reviewer" maxLength="41">Reviewer</string>
     <string name="show_deck_title" maxLength="41">Show deck title</string>
     <string name="accessibility" maxLength="41">Accessibility</string>
+    <string name="set_due_date" maxLength="41">Set due date</string>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/10-preferences.xml:184]: Please use the string from the backend

**Review by david-allison — CHANGES_REQUESTED** [review]

**Comment by Prince-kushwaha** [comment]
Hi @david-allison ,when trying to set the title of the "Set Due Date" ListPreference dynamically from the HeaderFragment. The problem occurs because you're trying to access the ListPreference, which is defined in preferences_controls.xml, while the HeaderFragment is loading its preferences from R.xml.preference_headers. Since the ListPreference isn't part of the current preference screen loaded by the HeaderFragment, it results in a null reference when you try to access it.

In other words, the fragment is only aware of the preferences defined in preference_headers.xml and doesn't have access to the preferences from preferences_controls.xml. That's why you're unable to dynamically set the title of the ListPreference("Set due date")
Screenshot 2024-10-08 134908

**Review by BrayanDSO — COMMENTED** [review]
This is 100% fixable while using the backend string. It would take me around 5 minutes to send a PR with the fix.

But since you made all of that fuss in #17158 and wanted all the credit without sending a PR, I'll leave to you to figure it out. That will be a good learning experience.

**Review by BrayanDSO — CHANGES_REQUESTED** [review]
forgot to press Request changes

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/HeaderFragment.kt [hunk 2]
```diff
@@ -127,6 +130,12 @@ class HeaderFragment : PreferenceFragmentCompat() {
                 index(R.xml.preferences_accessibility)
                 index(R.xml.preferences_backup_limits)
                 ignorePreference(activity.getString(R.string.pref_backups_help_key))
+                indexItem()
+                    .withKey(activity.getString(R.string.reschedule_command_key))
+                    .withTitle(setDuePreferenceTitle)
+                    .withResId(R.xml.preferences_controls)
+                    .addBreadcrumb(activity.getString(R.string.pref_cat_controls))
+                    .addBreadcrumb(setDuePreferenceTitle)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/preferences/HeaderFragment.kt:138]: This looks like the simplest way

`SearchIndexItem` doesn't seem extensible

A `Preference` can be passed into `indexItem`, but we don't have an instance of the subscreen here.

**Review by david-allison — APPROVED** [review]
Thanks

File: AnkiDroid/src/main/java/com/ichi2/anki/preferences/HeaderFragment.kt (hunk 2 above)
Comment (Prince-kushwaha) [reply AnkiDroid/src/main/java/com/ichi2/anki/preferences/HeaderFragment.kt:138]: yes

**Review by BrayanDSO — APPROVED** [review]
